### ISSUE TITLE
Clean up old /tmp/ctivo files after a crash

### ISSUE THREAD

**Issue by briancline** [dscottbuch/cTiVo#284]
If cTivo crashes, it leaves a fair number of files in `/tmp/ctivo` behind, including partially downloaded shows. If I instruct it to download the same show that it was downloading while it crashed, it doesn't seem to reuse the buffer file that was already there -- it creates a new one. Unsuspecting users may find they are gradually (or substantially, in the case of basketball games for instance) losing disk space to _something_ but might never know how what or where.

The obvious wishlist item here is that it would be great if it could pick back up where it left off on downloads (provided the TiVo provides some way to do so).
But... I'm very much a realist 😉 So it would at least be good if it could clean up these files at startup, or whenever.

Thinking out loud, maybe it could either ensure the directory is totally clean at startup, or alternatively use a subdirectory named after the PID of the main process such that when it crashes and starts back up, it can detect old/stale subdirs by comparing against its current PID. If it finds any, maybe either remove them, or at least pop open a dialog to ask to remove them and indicate how much space it'll free up? A few possibilities.

**Comment by mackworth** [comment]
It should completely clean /tmp/ctivo on startup (although not your own tmp if you change as discussed, as that’s too dangerous for us to assume, and people that do that should be aware of such things 🙃). If you’re not seeing then with 3.0, then do let me know.

And, no, TiVo’s file transfer does not support restarting a transfer.  Frankly, with the benign neglect we get from them, we’re grateful that we can still download shows.

**Comment by briancline** [comment]
Interesting. I saw it keeping around some older downloads and re-downloading the same recording to a different file even after I'd started it back up, so I manually cleaned up the old ones after verifying the mtimes weren't changing and lsof wasn't showing any open fds for the old ones. But tried to reproduce again a few minutes ago and it did seem to clear them at startup. Maybe I hit a strange edge case the first time.

I'll try with 3.x in the next week or so and see if it's still an issue. Thanks.
